Desktop_Interface/isobuffer.h

```diff
@@ -24,80 +25,89 @@ enum class UartParity : uint8_t;
 // Imagine it as a circular buffer, but with access functions specifically
 // designed for isochronous data from an Xmega.
 
-#define CONSOLE_UPDATE_TIMER_PERIOD (ISO_PACKETS_PER_CTX * 4)
+constexpr auto CONSOLE_UPDATE_TIMER_PERIOD = ISO_PACKETS_PER_CTX * 4;
 
 // TODO: Make private what should be private
+// TODO: Add m_ prefix to member variables
+// TODO: Change integer types to cstdint types
 class isoBuffer : public QWidget
 {
 	Q_OBJECT
 public:
+	// TODO: Add consoles as constructor arguments
 	isoBuffer(QWidget* parent = 0, int bufferLen = 0, isoDriver* caller = 0, unsigned char channel_value = 0);
 	// TODO?: Add a destructor
 
-	//Generic Functions
-	void openFile(QString newFile);
-
 //	Basic buffer operations
 	short bufferAt(int idx) const;
 	void insertIntoBuffer(short item);
 	void clearBuffer();
 	void gainBuffer(int gain_log);
-	void glitchInsert(short type); // NO-OP
 
 // Advanced buffer operations
+private:
+	template<typename T, typename Function>
+	void writeBuffer(T* data, int len, int TOP, Function transform);
+public:
 	void writeBuffer_char(char* data, int len);
 	void writeBuffer_short(short* data, int len);
 
+	// TODO: Change return value to unique_ptr
 	short* readBuffer(double sampleWindow, int numSamples, bool singleBit, double delayOffset);
 
 //	file I/O
 	bool maybeOutputSampleToFile(double convertedSample);
 	double sampleConvert(short sample, int TOP, bool AC) const;
 	short inverseSampleConvert(double voltageLevel, int TOP, bool AC) const;
 
+private:
+	template<typename Function>
+	int capSample(int offset, int target, double seconds, double value, Function comp);
+public:
 	int cap_x0fromLast(double seconds, double vbot);
 	int cap_x1fromLast(double seconds, int x0, double vbot);
 	int cap_x2fromLast(double seconds, int x1, double vtop);
-	void serialManage(double baudRate, int type, UartParity parity);
+	void serialManage(double baudRate, UartParity parity);
 
 // ---- MEMBER VARIABLES ----
 
 //	Presentantion?
 // NOTE: it seems like these are never initialized but they are used as though they were...
-	QPlainTextEdit* console1;
-	QPlainTextEdit* console2;
-	unsigned char channel = 255;
-	bool serialAutoScroll = true;
+	QPlainTextEdit* m_console1;
+	QPlainTextEdit* m_console2;
+	unsigned char m_channel = 255;
+	bool m_serialAutoScroll = true;
 
 // Conversion And Sampling
-	double voltage_ref = 1.65;
-	double frontendGain = (R4 / (R3 + R4));
-	int samplesPerSecond;
-	int sampleRate_bit;
+	double m_voltage_ref = 1.65;
```

Comment:
These 4 variables here are properties of the Labrador board and not this buffer, so I think it makes sense for them to be globals.